src/rdf4cpp/rdf/parser/ParsingError.hpp

```diff
@@ -25,41 +25,60 @@ struct ParsingError {
     uint64_t line;
     uint64_t col;
     std::string message;
+};
 
-    inline friend std::ostream &operator<<(std::ostream &os, ParsingError::Type const &err_t) noexcept {
-        switch (err_t) {
-            case Type::Internal:
-                os << "internal error";
-                break;
-            case Type::EofReached:
-                os << "unexpected end of file";
-                break;
-            case Type::BadSyntax:
-                os << "bad syntax";
-                break;
-            case Type::BadIri:
-                os << "bad iri";
-                break;
-            case Type::BadCurie:
-                os << "bad curie";
-                break;
-            case Type::BadLiteral:
-                os << "bad literal";
-                break;
-            case Type::BadBlankNode:
-                os << "bad blank node";
-                break;
-        }
+} // namespace rdf4cpp::rdf::parser
 
-        return os;
+namespace std {
+
+template<>
+struct formatter<::rdf4cpp::rdf::parser::ParsingError::Type> : formatter<string_view> {
```

Comment:
Why is this one inheriting from `std::formatter<std::string_view>`?
Same for ParsingError's formatter